### DISCUSSION TITLE
一点小小的疑问，关注注释方面

### DISCUSSION THREAD

**Discussion by ghost** [gkd-kit/gkd#789] [Q&A] [answered]
gkd的规则代码，支持插入注释语句吗？

目前试了一下方法，都会被软件本身删除掉注释语句

是gkd对规则文件有严格的字段验证机制，任何额外的字段都会被清理掉吗？

**Comment by lisonge** [comment, marked as answer]
> 支持插入注释语句吗？

不支持

---

规则本质只是 json，底层原理是 字符串反序列化为对象 -> 校验对象字段 ->  将对象重新转为字符串后保存

在这个过程中 注释/多余字段 均会在 反序列化为 过程中丢失

**Reply by ghost** [reply]
感谢

**Comment by AIsouler** [comment]
可以把注释写到desc字段里
